applications/nrf_desktop/src/modules/ble_bond.c

```diff
@@ -693,6 +697,65 @@ static int init(void)
 	return 0;
 }
 
+static bool ble_peer_event_handler(const struct ble_peer_event *event)
+{
+	if (state != STATE_ERASE_ADV) {
+		return false;
+	}
+
+	if (event->state == PEER_STATE_CONNECTED) {
+		/* Ensure that connected peer will have time to establish
+		 * security.
+		 */
+		if ((k_delayed_work_remaining_get(&timeout) < ERASE_ADV_NEW_CONN_TIMEOUT_MS) &&
```

Comment:
```
 * Note that this behavior is
 * inherently subject to race conditions with the pre-existing
 * timeouts and work queue, so care must be taken to synchronize such
 * resubmissions externally.
```
Do we handle this properly? I don't know this module well enough to tell.

Reply:
This won't be a problem here. This work is submitted only from the context of the work queue thread.